### ISSUE TITLE
Grouping on Entrytype with 

### ISSUE THREAD

**Issue by shivaway** [JabRef/jabref#8527]
### JabRef version

Latest development branch build (please note build date below)

### Operating system

Windows

### Details on version and operating system

Version 10.0.19044.1526

### Checked with the latest development build

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

1. Create a new group using "Searching by Keyword".
2. Use the Field Name of Entrytype
3. Use the search of Report
4. The group will find nothing, even though two items are there.
[test.zip](https://github.com/JabRef/jabref/files/8128476/test.zip)

### Appendix

...
<details>

<summary>Log File</summary>

```
Paste an excerpt of your log file here
```

</details>

**Comment by ThiloteE** [comment]
Have i not solved this here: https://github.com/JabRef/jabref/issues/8524 ? Is this the same problem @shivaway

**Comment by shivaway** [comment]
I'll try it tomorrow morning. Thank you!

Sent from Yahoo Mail on Android 
 
  On Wed, Feb 23, 2022 at 5:43 PM, ***@***.***> wrote:   


Have i not solved this here: #8524 ? Is this the same problem @shivaway ?

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by shivaway** [comment]
Just tried the development build.  The good thing is that it doesn't search the Title anymore; however the grouping with entrytype still has some issues.  I changed the entrytype to Protocol and it found all of them, but then I changed it to calibration and it still was finding Protocol.
Still something wacky going on.
Thank you!  This is a great program!
    On Wednesday, February 23, 2022, 06:44:44 PM CST, Shawn Way ***@***.***> wrote:  
 
 I'll try it tomorrow morning. Thank you!

Sent from Yahoo Mail on Android 
 
  On Wed, Feb 23, 2022 at 5:43 PM, ***@***.***> wrote:   


Have i not solved this here: #8524 ? Is this the same problem @shivaway ?

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by ThiloteE** [comment]
if you have an entry like this:

```
@calibration{test,
  entrytype = {protocol},
}
```

searching by keyword `protocol` will still find this entry.
searching by keyword `calibration` should not find this entry.

**Comment by shivaway** [comment]
I agree.  This is what I'm seeing.

the Report Group is :
When I choose it, I get:




    On Thursday, February 24, 2022, 10:38:01 AM CST, ThiloteE ***@***.***> wrote:  
 
 


if you have an entry like this:
@calibration{test,
  entrytype = {protocol},
}

searching by keyword protocol will still find this entry.
searching by keyword calibration should not find this entry.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by Siedlerchr** [comment]
You have to post images here on GitHub they are not visible when you answer via mail

**Comment by shivaway** [comment]
Sorry, I tried to post the following showing what was happening.
![1645734423318blob](https://user-images.githubusercontent.com/51136278/155608802-6af5beb1-ffdd-49ba-9ba0-5cfbc5cb7124.jpg)
![1645734464735blob](https://user-images.githubusercontent.com/51136278/155608806-8620798f-6be9-458e-9f85-72c699a3c0fc.jpg)
![1645734494055blob](https://user-images.githubusercontent.com/51136278/155608807-10d8bdfa-8181-4137-b33d-ebcdef57054f.jpg)

**Comment by ThiloteE** [comment]
So i take it this issue is solved?

**Comment by shivaway** [comment]
Not really.  If you look at the images, you can see that the Report group pulls up the Protocols.    On Thursday, February 24, 2022, 04:12:00 PM CST, ThiloteE ***@***.***> wrote:  
 
 


So i take it this issue is solved?

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by ThiloteE** [comment]
You know, hope is the greatest of all treasures 🤣

Anyway. i think it would be nice if you could upload the data of the entries here. They example file you uploaded in your first post holds different examples that are different to the pictures you show us.

**Comment by ThiloteE** [comment]
In the first example you uploaded none of the entries hold such a field:   `entrytype  = {Report},`

Therefore, the following statement is incorrect, because the items are NOT there:

> 4\. The group will find nothing, even though two items are there.
>         [test.zip](https://github.com/JabRef/jabref/files/8128476/test.zip)

As soon as i add `entrytype  = {Report},`, it will be added to the group.

I repeat, you are confusing entry type and entrytype. To make it more clear to you: One of them is a custom entry type defined under `options>customize entry types` and the other one is a custom field you added to this type of entry. Both are not the same. You also could have named the custom field `entrytypeX` or `XYZ` or `ANYTHING` and then put the keyword `Report` in there. So when you have a group that searches for the keyword `report` within a custom field that is called XYZ, it will find the entry and add it to the group, even though it will not look like it is the entrytype,

**Comment by shivaway** [comment]
I think I understand. I'll figure it out when I get into work tomorrow morning.

Sent from Yahoo Mail on Android 
 
  On Thu, Feb 24, 2022 at 4:49 PM, ***@***.***> wrote:   


In the first example you uploaded none of the entries hold such a field: entrytype = {Report},

Therefore, the following statement is incorrect, because the items are NOT there:


4. The group will find nothing, even though two items are there.
test.zip


As soon as i add entrytype = {Report},, it will be added to the group.

I repeat, you are confusing entry type and entrytype. To make it more clear to you: One of them is a custom entry type defined under options>customize entry types and the other one is a custom field you added to this type of entry. Both are not the same. You also could have named the custom field entrytypeX or XYZ or ANYTHING and then put the keyword Report in there. So when you have a group that searches for the keyword report within a custom field that is called XYZ, it will find the entry and add it to the group, even though it will not look like it is the entrytype,

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by ThiloteE** [comment]
![image](https://user-images.githubusercontent.com/73715071/155673841-524a05b5-ca82-48a2-ac79-828a4696479b.png)

**Comment by ThiloteE** [comment]
But i also have to say that you might be right that something is off there. 

Using your library from [here](https://github.com/JabRef/jabref/issues/8527#issue-1148686371):

1. When the keyword is report (hence within the group) and i change the keyword to `xyz` the entry gets removed from the group. Then when clicking UNDO (ctrl + z) three times, the entry is back in the group, but the keyword is still xyz, even though it should be report.

2. Even worse, when I change the keyword of the entry with citationkey `Calibration2021` from `protocol` to `xyz` (the entry gets removed from the group; This is correct), but then click on another entry, then back on that one entry where i just changed the keyword, the keyword IS BACK to what it was before i changed it (still `protocol`) in the required fields tab, but still `xyz` in `{} bibtex source` tab!

@shivaway, could you try to name the field entrytype differently e.g. customentrytypelabel or entrytypekeyword? Maybe it conflicts somewhere with the actual internal entry type after all?

**Comment by shivaway** [comment]
Absolutely. That was my plan, once I get back to work. Thanks for being so patient with me!

Sent from Yahoo Mail on Android 
 
  On Fri, Feb 25, 2022 at 2:05 AM, ***@***.***> wrote:   


But i also have to say that you might be right that something is off there.

Using your library from here:
   
   -    
When the keyword is report (hence within the group) and i change the keyword to xyz the entry gets removed from the group. Then when clicking UNDO (ctrl + z) three times, the entry is back in the group, but the keyword is still xyz, even though it should be report.

   -    
Even worse, when I change the keyword of the entry with citationkey Calibration2021 from protocol to xyz (the entry gets removed from the group; This is correct), but then click on another entry, then back on that one entry where i just changed the keyword, the keyword IS BACK to what it was before i changed it (still protocol) in the required fields tab, but still xyz in {} bibtex source tab!


@shivaway, could you try to name the field entrytype differently e.g. customentrytypelabel or entrytypekeyword? Maybe it conflicts somewhere with the actual internal entry type after all?

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by shivaway** [comment]
I think it does have the conflict.  I added a new field to each of the types called customentrytype and changed the reports to use customentrytype for the search and it now reports 5 instances.  It contains the two reports but still contains all the protocols, none of which have "Report" in any field.  In fact, that type doesn't even have the customentrytype field.  I'm uploading the current .bib
![Screenshot 2022-02-25 081101](https://user-images.githubusercontent.com/51136278/155729460-ba0969a8-b898-4b81-bd21-dcda678e27b1.png)
[test.zip](https://github.com/JabRef/jabref/files/8141784/test.zip)

Something weird is definitely going on.

**Comment by ThiloteE** [comment]
Just for the record, when i try to import your library, i get this message:

![image](https://user-images.githubusercontent.com/73715071/155742562-1661963d-3df2-4526-a3ab-9bd395076ef5.png)

**Comment by ThiloteE** [comment]
When i imported the library, all three of the entries with entry type Protocol, had `report` in customentrytype. Therefore, the second part of the following sentence is not true:
> It contains the two reports but still contains all the protocols, none of which have "Report" in any field.

**Comment by ThiloteE** [comment]
What we found out so far point to three things:

1. Jabref fails at reading and depicting the keyword from a customized entry field within the required fields entry editor tab under certain circumstances (see [this comment](https://github.com/JabRef/jabref/issues/8527#issuecomment-1050624479)) (i can reliably re
2. You could have misconfigured Jabref, because whenever i try to import your library, there is something missing and the library does not fit to what you say here on github.
3. If you configured Jabref correctly, it could mean that exporting/saving your library saves different data than what you expected.

1. and 2. is beyond my abilities. I could test a little bit more with regard to 3, but i really should work on my own paper right now xD

**Comment by ThiloteE** [comment]
Also, please change the name of this issue to something that makes this issue easier to find for people that have similar issues.

E.g.: Grouping by searching for keywords within customized fields

**Comment by shivaway** [comment]
Thank you very much! I'm not sure how I could misconfigure it from the menu's, but I'm willing to accept that.
    On Friday, February 25, 2022, 10:07:36 AM CST, ThiloteE ***@***.***> wrote:  
 
 


Also, please change the name of this issue to something that makes this issue easier to find for people that have similar issues.

E.g.: Grouping by searching for keywords within customized fields

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by ThiloteE** [comment]
Thanks Christoph, i might do some tests one of these days.

@shivaway check github, i uploaded a picture.
![image](https://user-images.githubusercontent.com/73715071/155774633-c1c802c9-7739-4805-9f18-fc25ee866e04.png)

**Comment by shivaway** [comment]
Interesting.  Looking at the source, I can see the error and have no idea why I put "Report" for Customentrytype.  What's even more fun is the non-customized entryetype has three items, when on the edit screen, I can see only one.
![Screenshot 2022-02-25 162001](https://user-images.githubusercontent.com/51136278/155810979-6184b234-fbd5-4f3e-b68f-eefcca60b3a3.jpg)

This is from the zip file I sent over.

Thanks again!

**Comment by shivaway** [comment]
Interesting. This was a newly created library from scratch to demo to some others.
I'll start over.
Thanks!

Sent from Yahoo Mail on Android 
 
  On Sat, Feb 26, 2022 at 2:35 AM, ***@***.***> wrote:   


I cannot reproduce neither the undo, nor the other phenomenon described in #8527 (comment) and also not the "one keyword in edit screen while there are three keywords in {} bibtex source" issue with a minimal working example within a fresh library (here, the library that does not reproduce), but i can reproduce it when i use the library you uploaded (here).

There seems to be a special factor at play that we seem to have failed to identify so far. Your library seems special.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by LIM0000** [comment]
From image in https://github.com/JabRef/jabref/issues/8527#issuecomment-1051125776, I believe this was not a misconfigure. 
Please read https://github.com/JabRef/jabref/issues/8911#issuecomment-1161239043. The image is the result of user "forcefully" assign entries into `WordKeywordGroup` via warning dialog asking users whether they would like to assign the original group's entries to this group.

![image](https://user-images.githubusercontent.com/49628911/174719067-e540157b-94fb-4687-9a75-be2b4dd6e90d.png)

**Comment by LIM0000** [comment]
@ThiloteE , for second issue from https://github.com/JabRef/jabref/issues/8527#issuecomment-1050624479, I could not find entry with citationkey Calibration2021 in library from [here](https://github.com/JabRef/jabref/issues/8527#issue-1148686371).
Could you please write reproduce steps for me to recreate the issue?
Thanks a lot.

**Comment by ThiloteE** [comment]
Ah my bad. I must have had automatic creation of citationkeys enabled back then.

I just tried again: I still can reproduce the issue for all entries that are in the group "Qualifications". Just go to the "other fields" tab, then look for "entrytype" and change `protocol` to `xyz`. Upon deleting one character, the entry will leave the group. So you have to go to "all entries" to finish the process. When you are done, click on another group or entry and you will see that the entrytype is still protocoll in the "other fields" tab, even though in `{} biblatex source` tab it will be xyz.

**Comment by LIM0000** [comment]
Thank you @ThiloteE , here is a way to replicate the second issue using fresh library.

1. Create a new library
2. Go to `Options` -> `Customize entry types`
3. Select `Article` and click on the drop down list. Please note that there is no `Entrytype` field available in the drop down list.

![111](https://user-images.githubusercontent.com/49628911/174788843-406ff727-5c5d-4bbf-b2f7-53b4e76ab26c.png)

4. Type in `Entrytype` and click on + button

![222](https://user-images.githubusercontent.com/49628911/174789297-ef1852c0-4d51-48c4-be17-0a30c2243549.png)
![333](https://user-images.githubusercontent.com/49628911/174789318-a96e18d6-52f2-4da5-9444-83a4ac54c48f.png)

5. Create 2 new articles
6. Navigate to `Optional fields`.

![555](https://user-images.githubusercontent.com/49628911/174789601-6cb663cb-cbac-400b-9b9a-d191fb48ccca.png)

7. Delete the value in `Entrytype`
8. Select second entry and reselect first entry
9. The `Article` value is back in `Entrytype` at `Optional fields`. However, it is being deleted in bibtex source.

<hr/>

My first guess to this issue is that there is an internal `EntryType` variable that does not allow user to modify it.
However, I could confirm that this second issue does not relate to #8527.

**Comment by ThiloteE** [comment]
Closing this.

It is **very very very likely** all issues here are caused by issues

#8917 and #8911

Thank you for your report @shivaway and your patience!
Special thanks to @LIM0000 for helping to debug this! :))

**Comment by shivaway** [comment]
thank you very much for looking into this!
    On Tuesday, June 21, 2022, 07:28:27 AM CDT, ThiloteE ***@***.***> wrote:  
 
 


Closed #8527 as completed.
